shapeshift/web
# feat: unify tx link generation logic

## Description

Use a generic helper function, `getTxLink`, that can take either a `SwapperName` or `Dex` enum value, and returns a TX URL. This let's us get a URL for any TX by passing it through a pure function.

This helper is now used by `useTxDetails` to ensure that the URLs in TX history use swapper-specific URLs (e.g. CoW Swap trades now link to `explorer.cow.fi` rather than `etherscan.io`).

Note, we don't currently pass enough information from `lib` for THORSwaps for this to work with THOR trades yet, but this change ensures that when we do update this that we'll correctly use `viewblock.io`.

## Notice


## Pull Request Type


## Issue (if applicable)

Contributes to https://github.com/shapeshift/web/issues/3446

## Risk

This affects all TX link generation. Small impact, small risk.

## Testing

- Ensure TX links on the trade screen for swaps on THORSwap, 0x, CowSwap, and Osmosis are as expected.
- Ensure CoW Swap trades now use `explorer.cow.fi` on the TX history page.

### Engineering

☝️ 

### Operations

☝️

## Screenshots (if applicable)

N/A
